newIDE/app/src/InstancesEditor/InstancesResizer.js

```diff
@@ -20,98 +61,247 @@ export default class InstancesResizer {
+      grabbingLocation === 'TopLeft' ||
+      grabbingLocation === 'BottomLeft' ||
+      grabbingLocation === 'Left';
+    const isTop =
+      grabbingLocation === 'TopLeft' ||
+      grabbingLocation === 'TopRight' ||
+      grabbingLocation === 'Top';
+
+    // Flip the deltas to end up in the nominal case of a bottom-right grabbed node.
+    // Because a negative deltaX on the left grabbing node will make the width greater
+    // as a positive deltaX on the right grabbing node would do.
+    const flippedTotalDeltaX = isLeft
+      ? -roundedTotalDeltaX
+      : roundedTotalDeltaX;
+    const flippedTotalDeltaY = isTop ? -roundedTotalDeltaY : roundedTotalDeltaY;
+
+    let scaleX =
+      (initialSelectionAABB.width() + flippedTotalDeltaX) /
+      initialSelectionAABB.width();
+    let scaleY =
+      (initialSelectionAABB.height() + flippedTotalDeltaY) /
+      initialSelectionAABB.height();
+
+    // Applying a rotation then a scaling can result to
+    // an affine transformation with a shear composite.
+    // So, keeping the aspect ratio ensures a transformation without any shear.
+    let hasRotatedInstance = false;
+    for (let i = 0; i < instances.length && !hasRotatedInstance; i++) {
+      // It only applies to instances that are not strait.
+      hasRotatedInstance = instances[i].getAngle() % 90 !== 0;
+    }
+    if (proportional || hasRotatedInstance) {
+      // Choose the axis where the selection is the biggest.
+      // That way the cursor is always on one edge.
+      if (
+        !canMoveOnY(grabbingLocation) ||
+        (canMoveOnX(grabbingLocation) &&
+          flippedTotalDeltaX * initialSelectionAABB.height() >
+            flippedTotalDeltaY * initialSelectionAABB.width())
+      ) {
+        scaleY = scaleX;
+      } else {
+        scaleX = scaleY;
+      }
+    }
+
+    // No negative scale because instances can't be mirrored
+    // Make the minimal selection size to 1 to avoid absorption
+    scaleX = Math.max(1 / initialSelectionAABB.width(), scaleX);
+    scaleY = Math.max(1 / initialSelectionAABB.height(), scaleY);
+
+    const anchorX =
+      initialSelectionAABB.right -
+      resizeGrabbingRelativePositions[grabbingLocation][0] *
+        initialSelectionAABB.width();
+    const anchorY =
+      initialSelectionAABB.bottom -
+      resizeGrabbingRelativePositions[grabbingLocation][1] *
+        initialSelectionAABB.height();
+
     for (let i = 0; i < instances.length; i++) {
       const selectedInstance = instances[i];
 
-      if (!selectedInstance.hasCustomSize()) {
+      let initialUnrotatedInstanceAABB = this._getOrCreateUnrotatedInstanceAABB(
+        selectedInstance
+      );
+      let initialInstanceOriginPosition = this._getOrCreateInstanceOriginPosition(
+        selectedInstance
+      );
+
+      const angle = ((selectedInstance.getAngle() % 360) + 360) % 360;
+      if (
+        !proportional &&
+        !hasRotatedInstance &&
+        (angle === 90 || angle === 270)
```

Comment:
Add a comment explaining why this if branch is specific to angle 90 and angle 270 (which can be understood by seeing you swap scaleX/Y in the setCustomWidth/Height, but needs a bit of time before it's clear, at least to me).